analysis/src/analyzer.js

```diff
@@ -2,65 +2,30 @@
 
 const Ana = require('./ana_log');
 
-const {Analyzer, generateAnalysis, FsUrlLoader, FsUrlResolver} = require('polymer-analyzer');
+const {Analyzer, generateAnalysis, FsUrlLoader, PackageUrlResolver} = require('polymer-analyzer');
 const pathlib = require('path');
 
-/**
- * Extends the FsUrlLoader to only read from the package directory, while
- * allowing reading of files in the parent of the package directory.
- */
-class UrlLoader extends FsUrlLoader {
-  /**
-   * @param root - package root
-   */
-  constructor(root) {
-    super(pathlib.join(root, '..'));
-
-    this.packageRoot = root;
-  }
-
-  /**
-   * Only return files from within the package directory.
-   */
-  async readDirectory(pathFromRoot, deep) {
-    const files = await new FsUrlLoader(this.packageRoot).readDirectory(pathFromRoot, deep);
-    const result = [];
-
-    // Filter out minified files etc.
-    for (const path of files) {
-      // eg. for file.min.js => file.js
-      const fileMatch = pathlib.basename(path).split('.');
-      if (fileMatch.length <= 2) {
-        result.push(path);
-        continue;
-      }
-
-      const strippedBaseName = fileMatch[0] + '.' + fileMatch[fileMatch.length - 1];
-      if (files.indexOf(pathlib.join(pathlib.dirname(path), strippedBaseName)) === -1) {
-        result.push(path);
-      }
-    }
-    return result;
-  }
-}
-
 class AnalyzerRunner {
   analyze(root, inputs) {
     return new Promise(async (resolve, reject) => {
       Ana.log('analyzer/analyze', inputs);
 
       var paths = inputs || [];
 
+      const isBower = paths.length > 0;
```

Comment:
how does this mean it's bower? (I don't know what inputs or paths are)

Reply:
Sorry, this is an implementation detail that's passed in from another place. In bower we have the "main" field, so we pass in the list of input files. In NPM we don't, so we analyze the whole package.

I've switched this around so this is now injected into this function, and this function operates differently depending on whether or not we said it was bower.